mmv1/third_party/terraform/services/apphub/data_source_apphub_discovered_service_test.go

```diff
@@ -0,0 +1,126 @@
+package apphub_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccDataSourceApphubDiscoveredService_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"org_id":          envvar.GetTestOrgFromEnv(t),
+		"random_suffix":   acctest.RandString(t, 10),
+		"billing_account": envvar.GetTestBillingAccountFromEnv(t),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testDataSourceApphubDiscoveredService_basic(context),
+			},
+		},
+	})
+}
+
+func testDataSourceApphubDiscoveredService_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(
+		`
+resource "google_project" "service_project" {
+	project_id ="apphub-sp-%{random_suffix}"
+	name = "Service Project"
+	org_id = "%{org_id}"
+  billing_account = "%{billing_account}"
+}
+
+resource "time_sleep" "wait_120s_for_service_project" {
+  depends_on = [google_project.service_project]
+  create_duration = "120s"
+}
+
+# Enable Compute API
```

Comment:
Is this not enabled by default? If you do definitely need this project service, you should be able to combine the sleep resource so that you aren't waiting twice. For example, create project, enable service, wait 120, then create your resource.

Reply:
No, it is not enabled by default. After creating project, we need to wait some time so that necessary IAM permissions get propagated. We can't enable compute api before this propogation. And enabling compute API also takes some time, so we need to wait there as well.

Reply:
done